sklearn/linear_model/tests/test_ransac.py

```diff
@@ -494,3 +494,17 @@ def test_ransac_fit_sample_weight():
     base_estimator = Lasso()
     ransac_estimator = RANSACRegressor(base_estimator)
     assert_raises(ValueError, ransac_estimator.fit, X, y, weights)
+
+
+def test_ransac_base_estimator_fit_sample_weight():
+    class DummyLinearRegression(LinearRegression):
```

Comment:
This estimator is exactly a LinearRegression then. So we could use the LinearRegression directly

Reply:
The sample_weight is optional in the original LinearRegression, the point of the dummy is to make it necessary in the call. The old _ransac.py code breaks, as it should, using this test.

Reply:
OK I see. I think the test below is better since we check the fitted model.